src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishReadyToRun.cs

```diff
@@ -411,9 +406,9 @@ private static bool IsTargetOsLinux(string runtimeIdentifier)
             else
                 publishDirectory.Should().NotHaveFile("System.Private.CoreLib.dll");
 
-            if (emitNativeSymbols && !IsTargetOsOsX(testProject.RuntimeIdentifier))
+            NuGetFramework framework = NuGetFramework.Parse(targetFramework);
+            if (emitNativeSymbols && (!IsTargetOsOsX(runtimeIdentifier) || framework.Version.Major >= 6))
```

Comment:
```suggestion
            if (emitNativeSymbols && (!IsTargetOsOsX(testProject.RuntimeIdentifier) || framework.Version.Major >= 6))
```
The passed `runtimeIdentifier` may be `null`, in which case it is inferred.

Reply:
Fixed in 9th commit, thanks Anton for pointing that out!